### PR TITLE
feat(policies): add EVO1 policy

### PR DESCRIPTION
**Pull request by imstevenpmwork**
## Title

feat(policies): add EVO1 policy

## Summary / Motivation

This PR adds **EVO1** ([MINT-SJTU/Evo-1](https://github.com/MINT-SJTU/Evo-1)) as a new LeRobot policy (`policy.type=evo1`). EVO1 is a Vision-Language-Action model that pairs an **InternVL3-1B** VLM backbone with a **flow-matching action head** that predicts continuous action chunks. The integration uses the native Hugging Face `transformers` InternVL implementation (`OpenGVLab/InternVL3-1B-hf`, no `trust_remote_code`) and plugs into the standard LeRobot training/eval/rollout stack, including **Real-Time Chunking (RTC)** for asynchronous inference. All changes are additive — no existing behavior is modified.

## Related issues

- Supersedes: #3545

## What changed

- New policy package `src/lerobot/policies/evo1/`: config, InternVL3 embedder, flow-matching action head, `Evo1Policy`, and pre/post processor pipelines. Supports stage-based finetuning (`stage1` = action head only, `stage2` = VLM + head).
- RTC support: `--inference.type=rtc` works with `lerobot-rollout`, adapting EVO1's flow convention to the shared `RTCProcessor`.
- Factory registration with lazy imports, new `cosine_annealing_with_warmup` scheduler, new `evo1` extra in `pyproject.toml` (`flash-attn` optional, detected at runtime).
- Docs: `docs/source/evo1.mdx` (install, training, inference, LIBERO eval) + policy README.
- No breaking changes — everything is additive.

## How was this tested (or how to run locally)

- **Tests added**: `tests/policies/evo1/test_evo1.py` — 29 CPU tests with a mocked InternVL backbone covering config validation, processor save/load round-trips, attention masking, empty-camera padding, `action_is_pad` loss masking, flow-matching integration, and RTC (kwarg guard rails, processor wiring, guidance pulling the chunk prefix toward the previous chunk, first-chunk equivalence with unguided sampling). Plus LIBERO env-processor dispatch tests in `tests/envs/test_dispatch.py`.
  ```bash
  uv run pytest tests/policies/evo1 tests/envs/test_dispatch.py -q
  ```
- **Manual checks**:
  - Stage 1 training on a real-robot dataset with `lerobot-train` (multi-GPU via accelerate).
  - Real-robot rollout of the trained checkpoint with both sync inference and `lerobot-rollout --inference.type=rtc`; RTC completes the task and yields smoother execution than sync.
  - A LIBERO checkpoint is training; benchmark numbers will be added to `docs/source/evo1.mdx` once available.

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`)
- [x] All tests pass locally (`pytest`)
- [x] Documentation updated
- [x] CI is green
- [x] Community Review: I have reviewed another contributor's open PR and linked it here: #

## Reviewer notes

- The flow-matching convention differs from pi0: EVO1 integrates t: 0→1 with `v = x₁ − x₀`. The RTC integration in `flow_matching.py::FlowmatchingActionHead.get_action` maps this onto the convention `RTCProcessor` expects (`time = 1 − t`, velocity sign flipped in both directions) — worth a close look. `test_flowmatching_rtc_guidance_pulls_prefix_toward_previous_chunk` pins the sign empirically.
- `processor_evo1.py::reconcile_evo1_processors` restores pipeline pieces that don't serialize (the `to_transition` converter and the action postprocessing step) when loading processors from a checkpoint; it is invoked from `factory.make_pre_post_processors`.
- Precision handling: the policy opens its own bf16 autocast on CUDA (independent of the trainer's AMP context) and computes the loss in fp32.
- `flash-attn` is deliberately not a dependency of the `evo1` extra; the embedder falls back to standard attention when it is missing.

TBD: update libero checkpoint + numbers

### CODE REVIEW

**Comment by imstevenpmwork** [comment]
@claude Review this PR

File: src/lerobot/policies/factory.py [hunk 1]
```diff
@@ -330,6 +337,14 @@ def make_pre_post_processors(
             revision=pretrained_revision,
         )
         _reconnect_relative_absolute_steps(preprocessor, postprocessor)
+        if isinstance(policy_cfg, Evo1Config):
+            from .evo1.processor_evo1 import ensure_evo1_processor_steps
+
+            preprocessor, postprocessor = ensure_evo1_processor_steps(
```
Comment (imstevenpmwork) [inline src/lerobot/policies/factory.py]: Can you check why this is necessary ?

**Comment by imstevenpmwork** [comment]
I wonder why we get this warning at training and inference time:

```
Loading weights: 100%|██████████████████████| 732/732 [00:00<00:00, 16212.61it/s]
InternVLModel LOAD REPORT from: OpenGVLab/InternVL3-1B-hf
Key            | Status     |  |
---------------+------------+--+-
lm_head.weight | UNEXPECTED |  |

Notes:
- UNEXPECTED:	can be ignored when loading from different task/architecture; not ok if you expect identical arch.
INFO 2026-07-02 12:02:00 matching.py:182 FlowmatchingActionHead num_inference_timesteps=32
```

**Comment by imstevenpmwork** [comment]
@claude Review this PR

**Comment by imstevenpmwork** [comment]
CI Full Test run: https://github.com/huggingface/lerobot/actions/runs/28679765130
